<launch>
  <!-- ARGUMENTS TO ALLOW RUN FROM ANYWHERE -->
  <arg name="home_dir"       default="$(find orp)" />
  <arg name="data_folder"    default="$(find orp)/apc_data" />
  <arg name="icp_list"       default="$(arg home_dir)/cfg/test2.list" />
  <arg name="cph_list"       default="$(arg home_dir)/cfg/test2.list" />
  <arg name="vfh_list"       default="$(arg home_dir)/cfg/test2.list" />
  <arg name="cvfh_list"      default="$(arg home_dir)/cfg/test2.list" />
  <arg name="openni"         default="false"/>
  <arg name="openni2"        default="false"/>
  <arg name="realsense"       default="false"/>
  <arg name="recognitionview" default="false"/>

  <rosparam command="load" file="$(arg home_dir)/cfg/sia5_demo.yaml" />
  <rosparam command="load" file="$(arg home_dir)/cfg/shelves_absolute.yaml" />

  <!-- DATA -->
  <group if="$(arg openni)">
    <include file="$(find openni_launch)/launch/openni.launch"> </include>
  </group>
  <group if="$(arg openni2)">
    <include file="$(find openni2_launch)/launch/openni2.launch">  </include>
  </group>
  <group if="$(arg realsense)">
    <include file="$(find realsense_camera)/launch/realsense_camera.launch"> </include>
  </group>
  <remap from="/camera/depth_registered/points" to="/camera/depth/points" />

  <node pkg = "amazon_challenge" type="load_item_params.py" name="$(anon orp_load_item_params)" args="$(find amazon_challenge)/scripts/APC.sqlite" output="screen" respawn="false" />

  <!-- CLASSIFICATION -->

  <node
    name    = "frame_broadcaster"  
    pkg     = "amazon_challenge"
    type    = "frame_broadcaster"
    args    = "base_link camera_link 0 0 0 0 0 0"

    respawn = "true"
    output  = "screen"
  />

  <node
    name    = "dummy_frame_broadcaster"  
    pkg     = "orp"
    type    = "dummy_frame_broadcaster"
    args    = "base_link camera_link 0 0 0 0 0 0"

    respawn = "true"
    output  = "screen"
  />

  <node
    name    = "segmentation"  
    pkg     = "orp"
    type    = "segmentation"
    args    = ""

    respawn = "true"
    output  = "screen"
  />

  <node
    name    = "icp_classifier"
    pkg     = "orp"
    type    = "icp_classifier"
    args    = "$(arg data_folder) $(arg icp_list)"

    respawn = "true"
    output  = "screen"
  />

  <!--<node
    name    = "cph_classifier"
    pkg     = "orp"
    type    = "cph_classifier"
    args    = "$(arg data_folder) $(arg cph_list)"

    respawn = "true"
    output  = "screen"
  /-->

  <!--<node
    name    = "vfh_classifier"
    pkg     = "orp"
    type    = "vfh_classifier"
    args    = "$(arg data_folder) $(arg vfh_list)"

    respawn = "true"
    output  = "screen"
  /-->

  <!--node
    name    = "cvfh_classifier"
    pkg     = "orp"
    type    = "cvfh_classifier"
    args    = "$(arg data_folder) $(arg cvfh_list)"

    respawn = "true"
    output  = "screen"
  /-->

  <!--node
    name    = "rgb_classifier"
    pkg     = "orp"
    type    = "rgb_classifier"
    args    = "$(arg data_folder) $(arg rgb_list)"

    respawn = "true"
    output  = "screen"
  /-->

  <!-- CONDUCTOR -->
  <node
    name    = "shelf_manager"
    pkg     = "orp"
    type    = "shelf_manager"

    respawn = "true"
    output  = "screen"
  />

  <node
    name    = "recognizer"
    pkg     = "orp"
    type    = "recognizer"
    args    = "$(arg cph_list)"

    respawn = "true"
    output  = "screen"
  />

  <!-- VISUALIZATION AND CONFIGURATION -->
  <node
    name    = "rqt_reconfigure"
    pkg     = "rqt_reconfigure"
    type    = "rqt_reconfigure"
    args    = ""

    respawn = "false"
    output  = "screen"
  />

  <group if="$(arg recognitionview)">
    <node
      name    = "rviz"
      pkg     = "rviz"
      type    = "rviz"
      args    = "-s $(arg home_dir)/rviz/splash.png"

      respawn = "false"
      output  = "screen"
    />
  </group>

</launch>
